nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -176,14 +176,38 @@ protected void flushKuduSession(final KuduSession kuduSession, boolean close, fi
         }
     }
 
-    protected KerberosUser loginKerberosKeytabUser(final String principal, final String keytab) throws LoginException {
-        final KerberosUser kerberosUser = new KerberosKeytabUser(principal, keytab);
+    protected KerberosUser loginKerberosKeytabUser(final String principal, final String keytab, ProcessContext context) throws LoginException {
+        final KerberosUser kerberosUser = new KerberosKeytabUser(principal, keytab) {
+            @Override
+            public synchronized boolean checkTGTAndRelogin() throws LoginException {
+                boolean didRelogin = super.checkTGTAndRelogin();
+
+                if (didRelogin) {
+                    final KerberosAction<KuduClient> kerberosAction = new KerberosAction<>(this, () -> buildClient(context), getLogger());
+                    AbstractKuduProcessor.this.kuduClient = kerberosAction.execute();
```

Comment:
Should this call `close()` on the old client before creating a new one? Is that safe to do in a multithreaded context?